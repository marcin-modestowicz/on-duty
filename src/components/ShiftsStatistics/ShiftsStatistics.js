//@flow
import React from "react";
import styles from "./ShiftsStatistics.scss";

type Props = {
  summary: Object[]
};

const ShiftsStatistics = ({ summary }: Props) => (
  <table className={styles.root}>
    <thead className={styles.header}>
      <tr>
        <th>Name</th>
        <th>Power</th>
        <th>High Preference</th>
        <th>High Preference Filled</th>
        <th>Low Preference</th>
        <th>Low Preference Filled</th>
        <th>Shifts</th>
        <th>Overall Shift Share</th>
        <th>Satisfaction</th>
      </tr>
    </thead>
    <tbody className={styles.body}>
      {summary.map(user => (
        <tr key={user.id}>
          <td>{user.name}</td>
          <td>{user.power}</td>
          <td>{user.highPreference}</td>
          <td>{user.highPreferenceFilled}</td>
          <td>{user.lowPreference}</td>
          <td>{user.lowPreferenceFilled}</td>
          <td>{user.shifts}</td>
          <td>{user.shiftsShare}</td>
          <td>{user.satisfaction}</td>
        </tr>
      ))}
    </tbody>
  </table>
);

export default ShiftsStatistics;
