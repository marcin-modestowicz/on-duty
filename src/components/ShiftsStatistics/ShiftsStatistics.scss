.root {
  border-collapse: collapse;
  border: 1px solid #4a4a4a;

  .header {
    font-size: .8em;
    color: #4a4a4a;
  }
  .body {
    font-size: .7em;

    tr {
      border-top: 1px solid #4a4a4a;
    }
  }

  th,
  td {
    padding: .5em;
    text-align: center;
  }
}
