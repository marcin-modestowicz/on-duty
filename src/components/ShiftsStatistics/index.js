//@flow
export { default } from "./ShiftsStatistics";
