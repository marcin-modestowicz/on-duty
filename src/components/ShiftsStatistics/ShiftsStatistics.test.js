// @flow
import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import ShiftsStatistics from "./ShiftsStatistics";

describe("ShiftsStatistics component", () => {
  let shiftsStatistics;

  beforeAll(() => {
    const summary = [
      {
        id: "abc1",
        name: "John Doe",
        power: 2,
        highPreference: 2,
        highPreferenceFilled: 1,
        lowPreference: 5,
        lowPreferenceFilled: 1,
        shifts: 4,
        shiftsShare: "20%",
        satisfaction: 0.1
      }
    ];
    shiftsStatistics = shallow(<ShiftsStatistics summary={summary} />);
  });

  test("should render", () => {
    expect(toJson(shiftsStatistics)).toMatchSnapshot();
  });
});
