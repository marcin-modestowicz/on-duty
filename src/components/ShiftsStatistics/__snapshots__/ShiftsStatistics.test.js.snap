// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShiftsStatistics component should render 1`] = `
<table
  className="root"
>
  <thead
    className="header"
  >
    <tr>
      <th>
        Name
      </th>
      <th>
        Power
      </th>
      <th>
        High Preference
      </th>
      <th>
        High Preference Filled
      </th>
      <th>
        Low Preference
      </th>
      <th>
        Low Preference Filled
      </th>
      <th>
        Shifts
      </th>
      <th>
        Overall Shift Share
      </th>
      <th>
        Satisfaction
      </th>
    </tr>
  </thead>
  <tbody
    className="body"
  >
    <tr
      key="abc1"
    >
      <td>
        John Doe
      </td>
      <td>
        2
      </td>
      <td>
        2
      </td>
      <td>
        1
      </td>
      <td>
        5
      </td>
      <td>
        1
      </td>
      <td>
        4
      </td>
      <td>
        20%
      </td>
      <td>
        0.1
      </td>
    </tr>
  </tbody>
</table>
`;
